internal/bgp/frr/config.go

```diff
@@ -50,56 +50,34 @@ hostname {{.Hostname}}
 ip nht resolve-via-default
 ipv6 nht resolve-via-default
 
-{{- range $pc := .PrefixesV4ForCommunity }}
-{{- range $p := $pc.Prefixes }}
-ip prefix-list {{$pc.Community}}-v4prefixes permit {{$p}}
-{{- end}}
-{{- end}}
-{{- range $pc := .PrefixesV6ForCommunity }}
-{{- range $p := $pc.Prefixes }}
-ip prefix-list {{$pc.Community}}-v6prefixes permit {{$p}}
-{{- end}}
-{{- end}}
-{{- range $pl := .PrefixesV4ForLocalPref }}
-{{- range $p := $pl.Prefixes }}
-ip prefix-list {{$pl.LocalPreference}}-v4localpref-prefixes permit {{$p}}
-{{- end}}
-{{- end}}
-{{- range $pl := .PrefixesV6ForLocalPref }}
-{{- range $p := $pl.Prefixes }}
-ip prefix-list {{$pl.LocalPreference}}-v6localpref-prefixes permit {{$p}}
-{{- end}}
-{{- end}}
 {{- range .Routers }}
 {{- range $n := .Neighbors }}
 route-map {{$n.Addr}}-in deny 20
-{{/* NOTE: it's possible to have global routes only because all the neighbors
-receive the same advertisements. Once that changes, we'll need prefix lists per neighbor */}}
-{{- range $.PrefixesV4ForLocalPref }}
-route-map {{$n.Addr}}-out permit {{counter $n.Addr}}
-  match ip address prefix-list {{.LocalPreference}}-v4localpref-prefixes
-  set local-preference {{.LocalPreference}}
-  on-match next
-{{- end }}
-{{- range $.PrefixesV6ForLocalPref }}
+{{- range $a := .Advertisements }}
+{{- range $pl := prefixesForLocalPref $a $n.IPFamily.String }}
+{{- range $p := $pl.Prefixes }}
+{{ipCmd $n.IPFamily.String}} prefix-list {{$n.Addr}}-{{$pl.LocalPreference}}-{{ipCmd $n.IPFamily.String}}-localpref-prefixes permit {{$p}}
+{{- end}}
 route-map {{$n.Addr}}-out permit {{counter $n.Addr}}
-  match ipv6 address prefix-list {{.LocalPreference}}-v6localpref-prefixes
-  set local-preference {{.LocalPreference}}
+  match {{ipCmd $n.IPFamily.String}} address prefix-list {{$n.Addr}}-{{$pl.LocalPreference}}-{{ipCmd $n.IPFamily.String}}-localpref-prefixes
```

Comment:
I wonder if we should hide names like
`{{$n.Addr}}-{{$pl.LocalPreference}}-{{ipCmd $n.IPFamily.String}}-localpref-prefixes` behind a function that takes the neighbour and the local pref.
Something like localPrefPrefixList (neighbour, localpref) 

Same for the others: communityPrefixListName() , allowedPrefixListName.
Wdyt?